packages/react-aria-components/src/Accordion.tsx

```diff
@@ -0,0 +1,150 @@
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {AriaDisclosureProps, useDisclosure} from '@react-aria/disclosure';
+import {ButtonContext} from './Button';
+import {ContextValue, Provider, RenderProps, SlotProps, useContextProps, useRenderProps} from './utils';
+import {DisclosureState, useDisclosureState} from '@react-stately/disclosure';
+import {forwardRefType} from '@react-types/shared';
+import {HoverEvents, useFocusRing} from 'react-aria';
+import {mergeProps, mergeRefs, useObjectRef} from '@react-aria/utils';
+import React, {createContext, DOMAttributes, ForwardedRef, forwardRef, ReactNode, useContext, useRef} from 'react';
+
+export interface AccordionItemProps extends Omit<AriaDisclosureProps, 'children' | 'contentRef'>, HoverEvents, RenderProps<AccordionItemRenderProps>, SlotProps {}
+
+export interface AccordionItemRenderProps {
+  /**
+   * Whether the accordion item is expanded (controlled).
+   * @selector [data-expanded]
+   */
+  isExpanded?: boolean,
+  /**
+   * Whether the accordion item has keyboard focus.
+   * @selector [data-focus-visible-within]
+   */
+  isFocusVisibleWithin: boolean,
+  /**
+   * Whether the accordion item is disabled.
+   * @selector [data-disabled]
+   */
+  isDisabled: boolean,
+  /**
+   * State of the accordion item.
+   */
+  state: DisclosureState
+}
+
+export const AccordionItemContext = createContext<ContextValue<AccordionItemProps, HTMLDivElement>>(null);
+export const DisclosureStateContext = createContext<DisclosureState | null>(null);
+
+interface InternalAccordionContextValue {
+  contentProps: DOMAttributes<HTMLElement>,
+  contentRef: React.RefObject<HTMLElement | null>
+}
+
+const InternalAccordionContext = createContext<InternalAccordionContextValue | null>(null);
+
+function AccordionItem(props: AccordionItemProps, ref: ForwardedRef<HTMLDivElement>) {
+  [props, ref] = useContextProps(props, ref, AccordionItemContext);
+  let state = useDisclosureState(props);
+  let contentRef = useRef<HTMLElement>(null);
+  let {triggerProps, contentProps} = useDisclosure({...props, contentRef}, state);
+  let {
+    isFocusVisible: isFocusVisibleWithin,
+    focusProps: focusWithinProps
+  } = useFocusRing({within: true});
+
+  let renderProps = useRenderProps({
+    ...props,
+    defaultClassName: 'react-aria-AccordionItem',
+    values: {
+      isExpanded: state.isExpanded,
+      isDisabled: props.isDisabled || false,
+      isFocusVisibleWithin,
+      state
+    }
+  });
+
+  return (
+    <Provider
+      values={[
+        [ButtonContext, {
+          slots: {
+            trigger: triggerProps
```

Comment:
Should we also add a `DEFAULT_SLOT` here so you can add additional buttons without errors?